ipld/test_helpers.go

```diff
@@ -66,3 +71,16 @@ func RandShares(t *testing.T, total int) []Share {
 
 	return shares
 }
+
+// CreateDAH creates DataAvailabilityHeader from the given shares and eds width
+func CreateDAH(ctx context.Context, shares [][]byte, size uint64, na format.NodeAdder) da.DataAvailabilityHeader {
+	batchAdder := NewNmtNodeAdder(
+		ctx,
+		format.NewBatch(ctx, na, format.MaxSizeBatchOption(int(size)*2)),
+	)
+	tree := wrapper.NewErasuredNamespacedMerkleTree(size/2, nmt.NodeVisitor(batchAdder.Visit))
+	attackerEDS, _ := rsmt2d.ImportExtendedDataSquare(shares, consts.DefaultCodec(), tree.Constructor)
+	_ = batchAdder.Commit()
+
+	return da.NewDataAvailabilityHeader(attackerEDS)
+}
```

Comment:
* We must not ignore the error from the Commit and Import!!!
* If you make it as a testhelper, then pass the `t` and check for errors with t. Pls never ignore errors
* There is no need to pass EDS size as we have it in the `shares` already = `len(shares)^2`
* Use ipld.DefaultCodec.
* There is already a constructor for DAH in core/da pkg, which makes one from the EDS.
  * Lets better make this func similar to the `AddShares` as they are basically the same besides that one does rsmt2d.Compute and this one rsmt2d.Import and probably not make it a testhelper but a regular func that returns an error.  So we should call it ImportShares, place it below AddShares and make it return EDS.

Reply:
done